mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenProjectConfigurator.java

```diff
@@ -58,18 +84,26 @@ void configureTestRunner(Model model) {
             failBecauseOfPluginVersionMismatch(model);
         }
 
-        dependencyResolver.addRequiredDependencies(model);
-
-        for (Plugin testRunnerPlugin : testRunnerPluginConfigurations) {
-            dependencyResolver.addAsPluginDependency(testRunnerPlugin);
-            final Object configuration = testRunnerPlugin.getConfiguration();
-            if (configuration != null) {
-                final Xpp3Dom configurationDom = (Xpp3Dom) configuration;
-                final Xpp3Dom properties = getOrCreatePropertiesChild(configurationDom);
-                properties.addChild(defineUsageMode());
-                properties.addChild(defineTestSelectionCriteria());
-            }
-        }
+        return testRunnerPluginConfigurations.stream()
+            .filter(
+                testRunnerPlugin -> !(testRunnerPlugin.getArtifactId().equals("maven-failsafe-plugin") && isSkipITs()))
+            .collect(Collectors.toList());
+    }
+
+    boolean isSkipTestExecutionSet() {
+        return isSkipTests() || isSkip();
+    }
+
+    private boolean isSkipTests() {
+        return Boolean.valueOf(System.getProperty("skipTests", "false"));
+    }
+
+    private boolean isSkipITs() {
+        return Boolean.valueOf(System.getProperty("skipITs", "false"));
+    }
+
+    private boolean isSkip() {
+        return Boolean.valueOf(System.getProperty("maven.test.skip", "false"));
```

Comment:
Please, move these methods outside of this class. The logic retrieves static information - so, they can be static methods. 
See this comment: https://github.com/arquillian/smart-testing/pull/102#discussion_r132418126